node/loop.go

```diff
@@ -337,27 +356,40 @@ func (t *loopingTransport) loop() {
 	_ = t.conn.Close()
 }
 
-func (t *loopingTransport) nextID() jsonrpc.ID {
+func (t *loopingTransport) nextID(seed jsonrpc.ID) jsonrpc.ID {
+	n := atomic.AddUint64(&t.counter, 1)
+	if seed.IsString {
+		return jsonrpc.ID{
+			Num:      0,
+			Str:      fmt.Sprintf("%s-%d", seed.Str, n),
+			IsString: true,
+		}
+	}
 	return jsonrpc.ID{
-		Num: atomic.AddUint64(&t.counter, 1),
+		Num: n,
 	}
 }
 
 func (t *loopingTransport) Request(ctx context.Context, r *jsonrpc.Request) (*jsonrpc.RawResponse, error) {
 
-	outbound := outboundRequest{
-		request:  r,
-		chResult: make(chan *jsonrpc.RawResponse),
-		chError:  make(chan error),
+	owned := &jsonrpc.Request{}
+	if err := copier.Copy(owned, r); err != nil {
```

Comment:
(Optional) If we wanted to avoid adding a dep, could probably do 

```go
owned.Params = append(make(Params, 0, len(r.Params)), r.Params...)
```

Reply:
I considered that, but felt that the copier approach a a little more self-documenting, at the expense of another dep (albeit a small and stable-looking one).